File: 3_tier_Project_01_NodeJS_Java_MySql/backend/src/main/java/com/devopsshack/projectops/entity/Project.java
package com.devopsshack.projectops.entity;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "projects")
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 120)
    private String name;

    @Column(nullable = false, length = 1000)
    private String description;

    @Column(nullable = false, length = 80)
    private String owner;

    @Column(nullable = false, length = 80)
    private String technology;

    @Column(nullable = false, length = 40)
    private String environment;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private ProjectStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Priority priority;

    @Column(nullable = false)
    private Integer progress;

    private LocalDate dueDate;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getOwner() { return owner; }
    public void setOwner(String owner) { this.owner = owner; }
    public String getTechnology() { return technology; }
    public void setTechnology(String technology) { this.technology = technology; }
    public String getEnvironment() { return environment; }
    public void setEnvironment(String environment) { this.environment = environment; }
    public ProjectStatus getStatus() { return status; }
    public void setStatus(ProjectStatus status) { this.status = status; }
    public Priority getPriority() { return priority; }
    public void setPriority(Priority priority) { this.priority = priority; }
    public Integer getProgress() { return progress; }
    public void setProgress(Integer progress) { this.progress = progress; }
    public LocalDate getDueDate() { return dueDate; }
    public void setDueDate(LocalDate dueDate) { this.dueDate = dueDate; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
}

File: 3_tier_Project_02_Pyton_DotNet_PostgreSQL/backend-dotnet/Models/Project.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectOps.Api.Models;

[Table("projects")]
public class Project
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required, MaxLength(120)]
    [Column("title")]
    public string Title { get; set; } = string.Empty;

    [Required, MaxLength(60)]
    [Column("category")]
    public string Category { get; set; } = string.Empty;

    [Required, MaxLength(30)]
    [Column("difficulty")]
    public string Difficulty { get; set; } = string.Empty;

    [Required, MaxLength(30)]
    [Column("status")]
    public string Status { get; set; } = string.Empty;

    [Required, MaxLength(80)]
    [Column("owner")]
    public string Owner { get; set; } = string.Empty;

    [Required, MaxLength(800)]
    [Column("description")]
    public string Description { get; set; } = string.Empty;

    [MaxLength(300)]
    [Column("repository_url")]
    public string? RepositoryUrl { get; set; }

    [Range(0, 100)]
    [Column("progress")]
    public int Progress { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
